### PR TITLE
Migrate MeshConfig Getters out of the configurator

### PR DESCRIPTION
**Pull request by steeling**
Move the MeshConfig getters onto the meshconfig itself, and out of the configurator class. This is part of [#4929](https://github.com/openservicemesh/osm/issues/4929), and will make it simpler to move the remaining methods into the k8s controller.

It makes the interface significantly smaller

### CODE REVIEW

**Review by shashankram — DISMISSED** [review]
I believe we discussed the idea of moving the Configurator implementation into the autogenerated API code part in the past, and I was against this for several reasons:
1. It results in extra maintenance, as every time a new API version is added/updated/removed, we have to manually copy over these non-autogenerated code over. I think it's best to keep the autogenerated code separate and keep custom wrapper implementations used by the controller within the controller code, and not `pkg/apis`.
2. This couples controller defaults and fallback behavior (on nil, empty strings, etc.) into the API code. I would argue this is confusing from an API perspective, as such defaults should be correctly encoded in the API. For e.g., when the init container image is unset, it returns `OSM_DEFAULT_INIT_CONTAINER_IMAGE`. This behavior is not intuitive from an API perspective. Such instances are better handled within the controller code.
3. I don't believe we need to expose multiple sub-fields via methods on a large type such as `MeshConfig`. Clients can directly reference the fields regardless of the nesting level. We should aim to reduce the number of accessors exposed. For e.g., the k8s `corev1.Pod` doesn't have unnecessary wrappers that client implementations may require. I would suggest following the k8s API style for the Go types as much as possible.

Can we first make a pass on reducing the number of getters on the Configurator type, and then see if having specific accessors makes sense?

**Comment by steeling** [comment]
> I believe we discussed the idea of moving the Configurator implementation into the autogenerated API code part in the past, and I was against this for several reasons:
> 
> 1. It results in extra maintenance, as every time a new API version is added/updated/removed, we have to manually copy over these non-autogenerated code over. I think it's best to keep the autogenerated code separate and keep custom wrapper implementations used by the controller within the controller code, and not `pkg/apis`.
> 2. This couples controller defaults and fallback behavior (on nil, empty strings, etc.) into the API code. I would argue this is confusing from an API perspective, as such defaults should be correctly encoded in the API. For e.g., when the init container image is unset, it returns `OSM_DEFAULT_INIT_CONTAINER_IMAGE`. This behavior is not intuitive from an API perspective. Such instances are better handled within the controller code.
> 3. I don't believe we need to expose multiple sub-fields via methods on a large type such as `MeshConfig`. Clients can directly reference the fields regardless of the nesting level. We should aim to reduce the number of accessors exposed. For e.g., the k8s `corev1.Pod` doesn't have unnecessary wrappers that client implementations may require. I would suggest following the k8s API style for the Go types as much as possible.
> 
> Can we first make a pass on reducing the number of getters on the Configurator type, and then see if having specific accessors makes sense?

done, moved remaining accessors to the utils package

File: pkg/k8s/events/types.go [hunk 1]
```diff
@@ -30,7 +39,145 @@ const (
 
 // PubSubMessage represents a common messages abstraction to pass through the PubSub interface
 type PubSubMessage struct {
-	Kind   announcements.Kind
+	Kind   Kind
+	Type   EventType
 	OldObj interface{}
 	NewObj interface{}
 }
+
```
Comment (shashankram) [inline pkg/k8s/events/types.go:47]: Are these event related changes accidentally included in this PR?

**Review by shashankram — COMMENTED** [review]
Could you drop the event related changes in this PR as I believe that is unrelated to the MeshConfig changes.

File: pkg/k8s/events/types.go (hunk 1 above)
Comment (steeling) [reply pkg/k8s/events/types.go:47]: Those got caught in here somehow. I'll  fix once https://github.com/openservicemesh/osm/pull/4939 is submitted

**Comment by steeling** [comment]
> Could you drop the event related changes in this PR as I believe that is unrelated to the MeshConfig changes.

rebased and ready to go!

**Review by keithmattix — APPROVED** [review]
Looks like this is functionally a no-op, so I'm approving

File: pkg/utils/mesh_config.go [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+const (
+	// defaultServiceCertValidityDuration is the default validity duration for service certificates
+	defaultServiceCertValidityDuration = 24 * time.Hour
+
+	// defaultIngressGatewayCertValidityDuration is the default validity duration for ingress gateway certificates
+	defaultIngressGatewayCertValidityDuration = 24 * time.Hour
+
+	// DefaultCertKeyBitSize is the default certificate key bit size
+	DefaultCertKeyBitSize = 2048
```
Comment (allenlsy) [inline pkg/utils/mesh_config.go:22]: nit: no need to export this?

**Review by allenlsy — APPROVED** [review]

File: pkg/utils/mesh_config.go (hunk 2 above)
Comment (steeling) [reply pkg/utils/mesh_config.go]: good call, fixed

**Review by shashankram — APPROVED** [review]

**Review by jaellio — APPROVED** [review]

**Pull request merged**